### DISCUSSION TITLE
Expose IDs for Islands

### DISCUSSION THREAD

**Discussion by natemoo-re** [withastro/roadmap#166] [Proposal]
> Hey question, do each island have a unique ID I could access inside a custom renderer? I want to isolate hydration instances in Solid's global hydration system and I realized Astro might already be generating a stable ID across both sides. Makes it easier to match things up. 

https://discord.com/channels/830184174198718474/916466736415539271/916466738013540413

**Comment by aFuzzyBear** [comment]
I think that having the ID's exposed to framework authors would hold them in good stead, esp if they are the likes of Ryan at Solid who has an understanding of this domain equal to that of a tenure professor
